spring-security-web-app/README.md

```diff
@@ -0,0 +1,225 @@
+
+#### Packaging and deploying to container
+
+If you would like to deploy the web sample to Tomcat, you will need to make a couple of changes to the source code.
+
+1. Open spring-security-web-app/pom.xml
+    - Under `<name>spring-security-web-app</name>` add `<packaging>war</packaging>`
+    - Add dependency:
+
+         ```xml
+         <dependency>
+          <groupId>org.springframework.boot</groupId>
+          <artifactId>spring-boot-starter-tomcat</artifactId>
+          <scope>provided</scope>
+         </dependency>
+         ```
+
+2. Open spring-security-web-app/src/main/java/com.microsoft.azure.springsecuritywebapp/SpringSecurityWebAppApplication
+
+    - Delete all source code and replace with the following:
+
+   ```Java
+        package com.microsoft.azure.springsecuritywebapp;
+
+        import org.springframework.boot.SpringApplication;
+        import org.springframework.boot.autoconfigure.SpringBootApplication;
+        import org.springframework.boot.builder.SpringApplicationBuilder;
+        import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;
+
+        @SpringBootApplication
+        public class SpringSecurityWebAppApplication extends SpringBootServletInitializer {
+
+         public static void main(String[] args) {
+          SpringApplication.run(SpringSecurityWebAppApplication.class, args);
+         }
+
+         @Override
+         protected SpringApplicationBuilder configure(SpringApplicationBuilder builder) {
+          return builder.sources(SpringSecurityWebAppApplication.class);
+         }
+        }
+   ```
+
+3. Open a command prompt, go to the root folder of the project, and run `mvn package`
+    - This will generate a `spring-security-web-app.war` file in your /targets directory.
+    - Rename this file to `ROOT.war`
+    - Deploy this war file using Tomcat or any other J2EE container solution.
+        - To deploy on Tomcat container, copy the .war file to the springsecuritywebapp's folder under your Tomcat installation and then start the Tomcat server.
+
+    This WAR will automatically be hosted at `http://<yourserverhost>:<yourserverport>/`
+        - Tomcats default port is 8080. This can be changed by
+        - Going to tomcat/conf/server.xml
+        - Search "Connector Port"
+        - Replace "8080" with your desired port number
+
+    Example: `http://localhost:8080`
+
+### You're done
+
+Click on "Login" to start the process of logging in. Once logged in, you'll see the name of the user who is authenticated. You'll then have the option to "Sign out".
+
+## About the code
+
+The relevant code for this sample is in the `AppConfiguration.java` file.
+
+This class extends `WebSecurityConfigurerAdapter` from which the `configure` method is overridden.
+
+```Java
+public void configure(HttpSecurity http) throws Exception {
```

Comment:
Can we also include the "@EnableOAuth2Sso" etc on top of the method since these are relevant to the auth working...

Reply:
Yes, included.